### DISCUSSION TITLE
Multi-threading in local dev builds?

### DISCUSSION THREAD

**Discussion by raziel2001au** [TheFutureGadgetsLab/WaifuXL#163] [Q&A] [answered]
Is there something I need to do to enable multi-threading in a locally running copy of WaifuXL? The officially hosted versions all properly multi-thread, but when I run a local version from source, it seems to be stuck running the processing in a single thread which is extremely slow.

Can anyone let me know what I'm missing?

Also, this is just a general comment, but a small amount of developer documentation would be really useful. It doesn't have to be anything complex, but just some general instructions on how to get up and running and what version of NodeJS is recommended. I'm not even sure if I'm using a supported version of NodeJS as I had to remove a whole bunch of stuff from next.config.js to get it running. The new de-spagetti branch runs fine without modifications, but both main and the de-spaghetti branch has the single-threading issue I mentioned when running locally with npm run start / rpm run dev.

**Comment by bensonalec** [comment, marked as answer]
So, it'll depend on what browser you're using. But, in Chrome you need to enable SharedArrayBuffee by running it with the flag --enable-features=SharedArrayBuffer. So, you'll want to launch it from the terminal and use that flag when you run chrome. I’ll go ahead and get some developer documentation for running it locally together later this evening. (Mobile formatting not working too well, so I apologize for this poorly formatted reply)

**Comment by raziel2001au** [comment]
Nice thanks, that works!

Some dev documentation would definitely be really awesome, I'm sure others will appreciate it as well. As a side note, I love what you guys have done with the project, it is such an amazing way to do it (without a full backend and everything).

**Reply by haydn-jones** [reply]
Thanks for the kind words, we're simply happy that we can provide a useful service.

Wrote some dev docs, lmk if this is sufficient or you would like some clarification: https://github.com/TheFutureGadgetsLab/WaifuXL/blob/main/README.md#running-locally
